File: tests_pmt_solver.py
import unittest
from pmt_solver import Loan

class TestPaymentSolver(unittest.TestCase):

    def test001(self):
        ''' years '''
        test_loan = Loan(100000,0,0.08,"years",25)
        test_pmt = test_loan.pmt()
        self.assertEqual(test_pmt, 9367.88)
    
    def test002(self):
        ''' months '''
        test_loan = Loan(100000,0,0.08,"months",25*12)
        test_pmt = test_loan.pmt()
        self.assertEqual(test_pmt, 771.82)
    
    def test003(self):
        ''' weeks '''
        test_loan = Loan(100000,0,0.08,"weeks",25*52)
        test_pmt = test_loan.pmt()
        self.assertEqual(test_pmt, 177.97)

    def test004(self):
        ''' days '''
        test_loan = Loan(100000,0,0.08,"days",25*365)
        test_pmt = test_loan.pmt()
        self.assertEqual(test_pmt, 25.35)

if __name__ == "__main__":
    unittest.main()

File: ruby/tests_pmt_solver.rb
require "minitest/autorun"
require_relative "pmt_solver"

class TestPaymentSolver < MiniTest::Unit::TestCase
    # def setup
    #     values = {  "beginning_value": 100000,
    #                 "future_value": 0,
    #                 "interest_rate": 0.08,
    #                 "payment_frequency": "years",
    #                 "periods": 25}
    #     @loan = Loan.new(values)
    # end

    def test_loan_one
        values = {  "beginning_value": 100000,
                    "future_value": 0,
                    "interest_rate": 0.08,
                    "payment_frequency": "years",
                    "periods": 25}
        loan1 = Loan.new(values)
        assert_equal 9367.88, loan1.pmt
    end

    def test_loan_two
        values = {  "beginning_value": 100000,
                    "future_value": 0,
                    "interest_rate": 0.08,
                    "payment_frequency": "months",
                    "periods": 25}
        loan2 = Loan.new(values)
        assert_equal 771.82, loan2.pmt
    end

end

